frontend/packages/dev-console/src/components/catalog/PinnedResourcesConfiguration.tsx

```diff
@@ -0,0 +1,344 @@
+        setPinnedResourcesConfigured(perspectiveDetails[index].pinnedResources);
+        setConfiguredPerspectives(consoleConfig?.spec?.customization?.perspectives);
+      }
+    }
+  }, [configuredPerspectives, consoleConfig, consoleConfigLoaded, defaultPins]);
+
+  const items = React.useMemo(() => {
+    return resources
+      .map((model: K8sKind) => {
+        return <Item title={model.labelKey ? t(model.labelKey) : model.kind} model={model} />;
+      })
+      .toArray();
+  }, [resources, t]);
+
+  const availableResources = React.useMemo<React.ReactElement<ItemProps>[]>(() => {
+    if (!consoleConfigLoaded) {
+      return [];
+    }
+    return items
+      ? items.filter(
+          (item) =>
+            !pinnedResources?.find(
+              (pinnedResource) =>
+                pinnedResource?.resource === item?.props?.model?.plural &&
+                (pinnedResource?.group === item?.props?.model?.apiGroup ||
+                  (!item?.props?.model?.apiGroup && pinnedResource?.group === '')) &&
+                pinnedResource?.version === item?.props?.model?.apiVersion,
+            ),
+        )
+      : [];
+  }, [consoleConfigLoaded, items, pinnedResources]);
+
+  const prePinnedResources = React.useMemo<React.ReactElement<ItemProps>[]>(() => {
+    if (!consoleConfigLoaded) {
+      return [];
+    }
+    let configuredResources = pinnedResources?.map((pinnedResource) => {
+      const itemForResource = items?.find(
+        (item) =>
+          pinnedResource?.resource === item?.props?.model?.plural &&
+          (pinnedResource?.group === item?.props?.model?.apiGroup ||
+            (!item?.props?.model?.apiGroup && pinnedResource?.group === '')) &&
+          pinnedResource?.version === item?.props?.model?.apiVersion,
+      );
+      return itemForResource || <InvalidItem title={pinnedResource?.resource} />;
+    });
+    configuredResources = configuredResources?.filter((element) => element !== undefined);
+    return configuredResources || [];
+  }, [consoleConfigLoaded, items, pinnedResources]);
+
+  const [saveStatus, setSaveStatus] = React.useState<SaveStatusProps>();
+  const save = useDebounceCallback(() => {
+    fireTelemetryEvent('Console cluster configuration changed', {
+      customize: 'Pre-pinned navigation items',
+      pinnedResources,
+    });
+    setSaveStatus({ status: 'in-progress' });
+    const patch: PerspectivesConsoleConfig = {
+      spec: {
+        customization: {
+          perspectives: perspectiveData,
+        },
+      },
+    };
+
+    patchConsoleOperatorConfig(patch)
+      .then(() => {
+        setPinnedResourcesConfigured(pinnedResources);
+        setSaveStatus({ status: 'successful' });
+      })
+      .catch((error) => {
+        setPinnedResources(pinnedResourcesConfigured);
+        setSaveStatus({ status: 'error', error });
+      });
+  }, 2000);
+
+  const onListChange = (
+    newEnabledOptions: React.ReactElement<ItemProps>[],
```

Comment:
do we need this ?

Reply:
From DualListSelector of Patternfly, onListChange 2 attributes are being passed, new options and old options.. and in all other places new option is kept simply or printed in logs if it is not used... So I kept it.